mem buffer holder created by allocator

weili1988 opened the pull request:
I created MemBufferHolder using allocator this time. Would you please take another look? I tried to makes sure that the mbh is created correctlly this time but the NullPointerException still exits. Thank you so much.

AI-Memory commented on mnemonic-core/src/test/java/org/apache/mnemonic/PersonBuffer.java:
```diff
@@ -0,0 +1,94 @@
+  public abstract MemBufferHolder getNamebuffer();
+
+  @DurableSetter
+  public abstract void setNamebuffer(MemBufferHolder mholder, boolean destroy);
```
This method didn't been used in the testcase, please check it.

AI-Memory commented on mnemonic-core/src/test/java/org/apache/mnemonic/DurablePersonBufferNGTest.java:
```diff
@@ -0,0 +1,167 @@
+  public void testGenPeople() throws OutOfHybridMemory, RetrieveDurableEntityError {
+    Random rand = Utils.createRandom();
+    NonVolatileMemAllocator act = new NonVolatileMemAllocator(Utils.getNonVolatileMemoryAllocatorService("pmalloc"),
+        1024 * 1024 * 8, "./pobj_person.dat", true);
```
Please use another name e.g. "./pobj_personbuffer.dat".

AI-Memory commented on mnemonic-core/src/test/java/org/apache/mnemonic/DurablePersonBufferNGTest.java:
```diff
@@ -0,0 +1,167 @@
+        for (int deep = 0; deep < rand.nextInt(100); ++deep) {
+
+          mother = PersonBufferFactory.create(act);
+          mother.setAge((short) (50 + rand.nextInt(50)));
```
Please insert "Assert.assertNotNull(mother);" to ensure not null.

weilicsu committed:
Mnemonic-97-debug

Adding new filed type to person

"NullPointerException"

Debug

mbh created by act
